central/graphql/resolvers/image_scan_test.go

```diff
@@ -0,0 +1,129 @@
+package resolvers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/graph-gophers/graphql-go"
+	imageCVEsDSMocks "github.com/stackrox/rox/central/cve/image/datastore/mocks"
+	imageDSMocks "github.com/stackrox/rox/central/image/datastore/mocks"
+	imageComponentsDSMocks "github.com/stackrox/rox/central/imagecomponent/datastore/mocks"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/testutils/envisolator"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	imageWithScanQuery = `
+		query getImages($query: String, $pagination: Pagination) {
+			images(query: $query, pagination: $pagination) { 
+				id
+				scan {
+					imageComponents {
+						name
+						imageVulnerabilities {
+							cve
+						}
+					}
+				}
+			}}`
+
+	imageWithoutScanQuery = `
+		query getImages($query: String, $pagination: Pagination) {
+			images(query: $query, pagination: $pagination) { 
+				id
+				imageComponents {
+					name
+					imageVulnerabilities {
+						cve
+					}
+				}
+			}}`
+)
+
+func TestImageScanResolver(t *testing.T) {
+	suite.Run(t, new(ImageScanResolverTestSuite))
+}
+
+type ImageScanResolverTestSuite struct {
+	suite.Suite
+
+	ctx         context.Context
+	envIsolator *envisolator.EnvIsolator
+	mockCtrl    *gomock.Controller
+
+	imageDataStore          *imageDSMocks.MockDataStore
+	imageComponentDataStore *imageComponentsDSMocks.MockDataStore
+	imageCVEDataStore       *imageCVEsDSMocks.MockDataStore
+
+	resolver *Resolver
+	schema   *graphql.Schema
+}
+
+func (s *ImageScanResolverTestSuite) SetupSuite() {
+	s.envIsolator = envisolator.NewEnvIsolator(s.T())
+	s.envIsolator.Setenv(features.PostgresDatastore.EnvVar(), "true")
+
+	if !features.PostgresDatastore.Enabled() {
+		s.T().Skip("Skip postgres store tests")
+		s.T().SkipNow()
+	}
+}
+
+func (s *ImageScanResolverTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+	s.ctx = contextWithImagePerm(s.T(), s.mockCtrl)
+
+	s.imageDataStore = imageDSMocks.NewMockDataStore(s.mockCtrl)
+	s.imageComponentDataStore = imageComponentsDSMocks.NewMockDataStore(s.mockCtrl)
+	s.imageCVEDataStore = imageCVEsDSMocks.NewMockDataStore(s.mockCtrl)
+
+	s.resolver, s.schema = setupResolver(s.T(), s.imageDataStore, s.imageComponentDataStore, s.imageCVEDataStore, nil)
+}
+
+func (s *ImageScanResolverTestSuite) TearDownTest() {
+	s.mockCtrl.Finish()
+}
+
+func (s *ImageScanResolverTestSuite) TearDownSuite() {
+	s.envIsolator.RestoreAll()
+}
+
+func (s *ImageScanResolverTestSuite) TestGetImagesWithScan() {
```

Comment:
@md2119 This tests fails with race detected

```
{Failed  ==================
WARNING: DATA RACE
Read at 0x00c001daa1c8 by goroutine 99:
  github.com/stackrox/rox/central/graphql/resolvers.(*imageResolver).Id()
      /go/src/github.com/stackrox/stackrox/central/graphql/resolvers/generated.go:6239 +0x4c
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:626 +0x48
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:339 +0xd7
  github.com/graph-gophers/graphql-go/internal/exec.execFieldSelection.func2()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:211 +0xa1b
  github.com/graph-gophers/graphql-go/internal/exec.execFieldSelection()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:231 +0x374
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelections.func1()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:81 +0x2f4
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelections·dwrap·2()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:82 +0x47

Previous write at 0x00c001daa1c8 by goroutine 100:
  github.com/stackrox/rox/central/graphql/resolvers.(*imageResolver).Scan()
      /go/src/github.com/stackrox/stackrox/central/graphql/resolvers/images.go:428 +0x1d7
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:626 +0x48
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:339 +0xd7
  github.com/graph-gophers/graphql-go/internal/exec.execFieldSelection.func2()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:211 +0xa1b
  github.com/graph-gophers/graphql-go/internal/exec.execFieldSelection()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:231 +0x374
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelections.func1()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:81 +0x2f4
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelections·dwrap·2()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:82 +0x47

Goroutine 99 (running) created at:
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelections()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:77 +0x212
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelectionSet()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:267 +0x478
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execList.func1()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:332 +0x387
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execList·dwrap·5()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:333 +0x47

Goroutine 100 (running) created at:
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelections()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:77 +0x212
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execSelectionSet()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:267 +0x478
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execList.func1()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:332 +0x387
  github.com/graph-gophers/graphql-go/internal/exec.(*Request).execList·dwrap·5()
      /go/pkg/mod/github.com/graph-gophers/graphql-go@v1.3.0/internal/exec/exec.go:333 +0x47
==================
    testing.go:1152: race detected during execution of test}
```
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/stackrox_stackrox/2720/pull-ci-stackrox-stackrox-master-go-unit-tests/1560194439421366272